### PR TITLE
feat: useSnap hook, connected address

### LINKED ISSUE #34: example: connect address

### PR DESCRIPTION
**Pull request by irubido**
handles users not having metamask or metamask flask
useSnap hook
Closes #34

### CODE REVIEW

File: packages/example/src/hooks/useSnap.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import {
+  isSnapInstalled,
+  isMetaMaskFlaskAvailable,
+  AleoSnap,
+} from '@chainsafe/aleo-snap-adapter';
+import { MetamaskRpcRequest } from '@chainsafe/aleo-snap-shared';
+import { useCallback, useEffect, useState } from 'react';
+
+const isDev = true;
```
Comment (BeroBurny) [inline packages/example/src/hooks/useSnap.ts:9]: Check out https://vitejs.dev/guide/env-and-mode.html#env-variables for `import.meta.env.DEV`

File: packages/example/src/components/AddSnap/AddSnap.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  useEffect(() => {
+    void (async () => {
+      return await new Promise<void>(() =>
+        setTimeout(() => {
+          setInitialSpinner(false);
+          return;
+        }, 1000),
+      );
+    })();
+  }, []);
```
Comment (BeroBurny) [inline packages/example/src/components/AddSnap/AddSnap.tsx:22]: maybe better to move this logic inside a snap, and add states like `isReady` for this kind of spinner

File: packages/example/src/hooks/useSnap.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      return;
+    }
+    void (async () => {
+      const isMetaMaskFlask = await isMetaMaskFlaskAvailable();
+      if (!isMetaMaskFlask) return;
+      const isInstalled = await isSnapInstalled(snapOrigin, version);
+      setIsMetaMaskFlask(isMetaMaskFlask);
+      setSnapInstalled(isInstalled);
+    })();
+  }, [snapInstalled]);
```
Comment (BeroBurny) [inline packages/example/src/hooks/useSnap.ts:66]: looks like it is better to split in 2 separate effects

**Review by BeroBurny — CHANGES_REQUESTED** [review]

File: packages/example/.env.sample [hunk 4]
```diff
@@ -0,0 +1,4 @@
+VITE_APP_IS_DEV=false
+VITE_APP_SNAP_ORIGIN_LOCAL='local:http://localhost:8081'
+VITE_APP_SNAP_ORIGIN_NPM='npm:@chainsafe/aleo-snap'
+VITE_APP_SNAP_VERSION_NPM='1.0.5'
```
Comment (BeroBurny) [inline packages/example/.env.sample:4]: please do not use hardcoded version

File: packages/example/.env.sample [hunk 5]
```diff
@@ -0,0 +1,4 @@
+VITE_APP_IS_DEV=false
```
Comment (BeroBurny) [inline packages/example/.env.sample:1]: you have env that can help you with that!

**Review by BeroBurny — CHANGES_REQUESTED** [review]

**Comment by irubido** [comment]
- vite dev always uses local snap origin

- vite build builds with npm snap origin
- vite build:local builds with local snap origin

**Review by BeroBurny — APPROVED** [review]

**Pull request merged**